### DISCUSSION TITLE
useServerInsertedHTML callback injected twice

### DISCUSSION THREAD

**Discussion by Nedilko** [vercel/next.js#43551] [Help]
I tried to use stitches.dev with next 13.

Added wrapper around all body content in root layout file.
Inside wrapper used hook `useServerInsertedHTML` with callback which returns style tag.
But **sometimes** this style tag injects not only in head, but also in body!

**Behaviour when style tag adds to body reproduces not always.**

Also I haven't found any descriptive documentation on `useServerInsertedHTML` hook.
Any suggestions how to make behaviour stable?

**layout.tsx:**
```typescript
import { ServerStylesheet } from './ServerStylesheet';

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang="en">
      {/*
        <head /> will contain the components returned by the nearest parent
        head.tsx. Find out more at https://beta.nextjs.org/docs/api-reference/file-conventions/head
      */}
      <head />
      <body>
        <ServerStylesheet>{children}</ServerStylesheet>
      </body>
    </html>
  );
}
```

**ServerStylesheet.tsx**

```typescript
"use client"

import { useServerInsertedHTML } from "next/navigation"
import { getCssText } from "../stitches.config"

type ServerStylesheetProps = {
  children: any,
}

export function ServerStylesheet({ children }: ServerStylesheetProps) {
  useServerInsertedHTML(() => {
    return <style id="stitches" dangerouslySetInnerHTML={{ __html: getCssText() }} />
  })

  return children
}
```

<img width="651" alt="Screenshot 2022-11-30 at 01 47 30" src="https://user-images.githubusercontent.com/15196815/204673442-5f99f366-6082-4270-9128-e40602895f32.png">

**Comment by Nedilko** [comment]
anyone tried to use useServerInsertedHTML and render style?

**Reply by Nedilko** [reply]
Any suggestions here?

**Reply by Nedilko** [reply]
any suggestions here?

**Comment by eljose47** [comment]
this seems to be the same problem and got answered:
https://github.com/vercel/next.js/discussions/49354#discussioncomment-6279917
